### ISSUE TITLE
Project versioning

### ISSUE THREAD

**Issue by Superbelko** [godot-dlang/godot-dlang#132]
Ok we have godot-4.0 in master, and we have godot-4.1 branch, next week I probably start to work on 4.2 since it is expected to reach beta stage, obviously we don't have enough maintainers for all versions.

So the problem is...
- How we should deal with godot 4.0 support (current master)? 
- Should we just forget it and merge godot-4.1 into master? 
- Or we should have separate branch for each minor version?

Maintaining multiple branches is a bit stretching our resources, also godot 4.1 and up introduced extension versioning support and it is supposed to be backward compatible, so I think we might say goodbye to godot 4.0 and focus on supporting newer versions.
What do you think guys?

*(This decision can be delayed to first 4.2 updates to see how it will work when dealing with older extension in newer godot version.)*

**Comment by al1-ce** [comment]
On one hand each minor version is intended to be updated to, but on other hand judging by gd3 there isn't gonna be much minor version updates, there were only 5, but I'd expect more for gd4.

Perhaps let's have a branch for each minor version, ignoring patches (aka 4.1.X -> X part), and keep current (would be 4.2 rn) version on main and create branch on new minor version release

**Comment by al1-ce** [comment]
TLDR:
- Branches same as godot repo
- Support for old versions dropped since everyone would update engine anyway

@Superbelko I've thinked about it and I suppose it's best to do branches same as godotengine/godot 

![image](https://github.com/godot-dlang/godot-dlang/assets/60989682/a8eee12b-a7b3-49ae-849d-c81a916d82f3)

So, that's gonna be that development happens in master branch and as we go we push it into their own branches (it's only logical to do dev on master if there is master), when there's gonna be 4.3 only then we gonna do 4.2 branch. Plus if we look at godot-cpp they're doing basically same thing as I'm proposing, only difference is that they target master (which would be latest released beta/rc in our case)

![image](https://github.com/godot-dlang/godot-dlang/assets/60989682/766d1bf3-e086-4993-8185-9be63cd9f365)

Also on note of support for certain versions: minor godot versions are supposed to be updated to when they come out and there's no real purpose in maintaining for previous version, unless it's some sort of a fork which we wouldn't support anyway

Also it seems godot invented some backwards compatibility stuff

Anyway I'm gonna close this issue now and do minor branch "orgranisation" according to this comment

**Comment by Superbelko** [comment]
Ok that's cool and all, but how do i revert a faulty feature now? Ok i can do a revert of a single file change, but what's about 40 files?
How do I merge that one specific feature to another branch?
In future please avoid squashing commits unless absolutely necessary, history is there for a reason...

**Comment by al1-ce** [comment]
Sorry, was a default choice on merge button, didn't notice
